### ISSUE TITLE
docker stack deploy does not recognize rw mount mode

### ISSUE THREAD

**Issue by teohhanhui** [docker/docker#32398]
**Description**

#30597 introduced a regression when it comes to mount with `:rw` suffix. It should be valid according to https://docs.docker.com/engine/reference/run/#volume-shared-filesystems which says:

> If neither 'rw' or 'ro' is specified then the volume is mounted in
read-write mode.

**Steps to reproduce the issue:**
1. `docker stack deploy -c docker-stack.yml mystack`

docker-stack.yml:
```yaml
version: '3.0'

services:
  nginx:
    image: nginx:1.11-alpine
    volumes:
      - nginx-configs:/etc/nginx/conf.d:rw
      - ./vhost.d:/etc/nginx/vhost.d:ro
      - nginx-contents:/usr/share/nginx/html:rw
      - nginx-logs:/var/log/nginx:rw

volumes:
  nginx-configs: {}
  nginx-contents: {}
  nginx-logs: {}
```

**Describe the results you received:**
> 3 error(s) decoding:
>
> * invalid spec: nginx-configs:/etc/nginx/conf.d:rw: unknown option: rw
> * invalid spec: nginx-contents:/usr/share/nginx/html:rw: unknown option: rw
> * invalid spec: nginx-logs:/var/log/nginx:rw: unknown option: rw

**Describe the results you expected:**
> Creating network mystack_default
> Creating service mystack_nginx

**Additional information you deem important (e.g. issue happens only occasionally):**
N/A

**Output of `docker version`:**

```
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:07:42 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:07:42 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 22
 Running: 8
 Paused: 0
 Stopped: 14
Images: 245
Server Version: 17.04.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 356
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: j0bscfxu5pwuhjs2uqa37tlze
 Is Manager: true
 ClusterID: y9d829wmrr8vg2pso3yaptb2i
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.1.114
 Manager Addresses:
  192.168.1.114:2377
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-71-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.711GiB
Name: YOGA
ID: M77W:GFMI:MAJO:7JRJ:4Z7V:3ZV2:MQDT:XKEB:LBIJ:6SIU:PKCU:2354
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: teohhanhui
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical

**Comment by thaJeztah** [comment]
ping @vdemeester @dnephin PTAL

**Comment by dnephin** [comment]
Thanks for the bug report. I've got a fix in #32407

As a workaround you can just remove the `:rw` since that is always the default.
